.github/CONTRIBUTING.md

```diff
@@ -17,40 +16,103 @@ We prefer questions and support requests be posted in the [forum](http://forum.t
+
+Samples: (even more [samples](https://github.com/angular/angular/commits/master))
+
+```
+docs(changelog): update changelog to beta.5
+```
+```
+fix(release): need to depend on latest rxjs and zone.js
+
+The version in our package.json gets copied to the one we publish, and users need the latest of these.
+```
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: gradle, git, idea)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Jenkins)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix for the game
+* **perf**: A code change that improves performance of the game
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, copyright, etc)
+* **test**: Adding missing tests or correcting existing tests
+
+### Scope
+The scope should be the name of the package(s) affected (as perceived by the person reading the changelog generated from commit messages).
+The package prefix `org.terasology` should be ommitted.
 
-1. If your work spans multiple local commits make sure they can be merged directly into the target branch. If you have some commit and merge noise in the history have a look at git-rebase. Ideally, every commit should be able to be used in isolation -- that is, each commit must build and pass all tests.
-1. The first line of the commit message should be a descriptive sentence about what the commit is doing. It should be possible to fully understand what the commit does by just reading this single line.
-1. **Optional:** Following the single line description (ideally no more than 70 characters long) should be a blank line followed by a list with the details of the commit.
-1. **Optional:** Add `Fix/Fixing/Fixes/Close/Closing/Refs #issue` if you want to mark the issue mentioned as complete in the issue tracker when the change is merged.
+Samples:
+```
+feat(world): add JOML API to WorldProvider
+```
+
+There are currently a few exceptions to the "use package name" rule:
+
+> :construction: What are our exceptions? Non-code changes (build, ci, ...)?
+
+### Subject
+The subject contains a succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* don't capitalize the first letter
+* no dot (.) at the end
+
+### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+### Footer
```

Comment:
Even though my grasp of English _is_ pretty close to native, I'm mildly confused here between subject body and footer 😅 

I figured Subject is essentially the first sentence / line, some tools like Git Kraken has a separate line for that. Then Body is the rest of the message. Is Footer just meant to be a general reference to the _bottom_ line of a commit message, or is it something more formal I'm just not familiar with?

Reply:
I think it's more or less the bottom line (except, it can also be multiple lines). In the Angular project they use a double empty line to separate it, I think. The whole point here is that all references to issues or PRs that are closed/fix should be in one place at the bottom.

**"Good"**
```
refactor!: adjust the API of FancyEvent

This migrates the API to work with S instead of T, and renames it accordingly.

BREAKING: FancyEvent.foo was renamed to FancyEvent.bar
Closes #123
```

**Bad**
```
refactor!: adjust the API of FancyEvent

This migrates the API to work with S instead of T (BREAKING). We close #123 this way.

Renames the API methods accordingly.
```

As we don't do full automation on this I don't think it is important that we define the technical details, but having the important information that can't be expressed in the subject in a single, easy to find place seems like a good thing to me.